src/FileCache.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace Aws;
+
+class FileCache implements CacheInterface
+{
+    private $cacheDir;
+
+    public function __construct()
+    {
+        $this->cacheDir = (new JsonCompiler)->getCacheDir();
+    }
+
+    public function set($key, $data, $ttl = 0)
+    {
+        $toSave = ['data' => $data];
+        $path = $this->getCachePath($key);
+        if ($ttl) {
+            $toSave['expiration'] = time() + $ttl;
+        }
+
+        if (@mkdir(dirname($path), 0777, true)) {
+            file_put_contents($path, \serialize($toSave), LOCK_EX);
+        }
+    }
+
+    public function remove($key)
+    {
+        unlink($this->getCachePath($key));
+    }
+
+    public function get($key)
+    {
+        $path = $this->getCachePath($key);
+
+        if (file_exists($path) && $retrieved = @file_get_contents($path)) {
+            $cached = \unserialize($retrieved);
+            if (isset($cached['data']) &&
+                (empty($cached['expiration']) || $cached['expiration'] > time())
+            ) {
+                return $cached['data'];
+            }
+
+            $this->remove($key);
+        }
+
+        return null;
```

Comment:
You're right. I had it in as documentation (to signal that null is the intentional return value for this code path), but the docblock already states that the method can return null.